components/ButtonFooter.js

```diff
@@ -15,49 +15,61 @@
  */
 
 import React, { Component } from 'react'
-import ReactNative, { TouchableOpacity, Text, View, Linking } from 'react-native'
+import ReactNative, { TouchableOpacity, Text, View, Linking, Alert } from 'react-native'
 import client, {Color} from '@doubledutch/rn-client'
 
 export default class ButtonFooter extends Component {
   constructor(props) {
     super(props)
   }
-    render(){
-      const { buttons } = this.props
-      return(
-        <View style={s.container}>
-          <TouchableOpacity onPress={()=>{
-            this.openWebURL(buttons[0].buttonURL)
-          }}>
-            <View style={s.buttonBox}>
-              <Text style={s.button}>{buttons[0].buttonTitle}</Text>
-            </View>
-          </TouchableOpacity>
-          <TouchableOpacity onPress={()=>{
-            Linking.openURL(buttons[1].buttonURL)
-          }} style={{marginTop:20}}>
-            <View style={s.buttonBox}>
-              <Text style={s.button}>{buttons[1].buttonTitle}</Text>
-            </View>
-          </TouchableOpacity>
-        </View>
-      )
+  render(){
+    const { buttons } = this.props
+    return(
+      <View style={s.container}>
+        <TouchableOpacity onPress={()=>{
+          this.linkCheck(this.linkCheck(buttons[0].buttonURL))
+        }}>
+          <View style={s.buttonBox}>
+            <Text style={s.button}>{buttons[0].buttonTitle}</Text>
+          </View>
+        </TouchableOpacity>
+        <TouchableOpacity onPress={()=>
+          this.linkCheck(buttons[1].buttonURL)
+        } style={{marginTop:20}}>
+          <View style={s.buttonBox}>
+            <Text style={s.button}>{buttons[1].buttonTitle}</Text>
+          </View>
+        </TouchableOpacity>
+      </View>
+    )
+  }
+
+  linkCheck = (link) => {
```

Comment:
Since this function is copied in multiple places, let's export it from a single file and import it from every component where it is needed